src/metabase/api/search.clj

```diff
@@ -512,6 +517,7 @@
         total-results       (cond->> (scoring/top-results reducible-results search.config/max-filtered-results xf)
                               true hydrate-user-metadata
                               (:model-ancestors? search-ctx) (add-dataset-collection-hierarchy)
+                              true (map add-can-write)
```

Comment:
Do you think we can move this map operation into the `xf` above?

Reply:
Good idea, done!